src/main/java/com/kosherjava/zmanim/AstronomicalCalendar.java

```diff
@@ -832,9 +832,11 @@ public Object clone() {
 		} catch (CloneNotSupportedException cnse) {
 			// Required by the compiler. Should never be reached since we implement clone()
 		}
-		clone.setGeoLocation((GeoLocation) getGeoLocation().clone());
-		clone.setCalendar((Calendar) getCalendar().clone());
-		clone.setAstronomicalCalculator((AstronomicalCalculator) getAstronomicalCalculator().clone());
+        if (clone != null) {
+			clone.setGeoLocation((GeoLocation) getGeoLocation().clone());
+			clone.setCalendar((Calendar) getCalendar().clone());
+			clone.setAstronomicalCalculator((AstronomicalCalculator) getAstronomicalCalculator().clone());
+		}
 		return clone;
 	}
```

Comment:
This was just to get rid of a possible NullPointerException warning. I know it is not necessary, but it is still beneficial to get rid of a few warnings.